src/Microsoft.AspNetCore.SignalR.Client.Core/IConnectionFactory.cs

```diff
@@ -1,15 +1,18 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Connections;
 
 namespace Microsoft.AspNetCore.SignalR.Client
 {
     public interface IConnectionFactory
     {
-        Task<ConnectionContext> ConnectAsync(TransferFormat transferFormat);
+        Task<ConnectionContext> ConnectAsync(TransferFormat transferFormat, CancellationToken cancellationToken = default);
```

Comment:
I'm not sure if it should have a default on interfaces. Should look and see what async APIs with interfaces do in corefx.

Reply:
I think it's fine, `Foo(int bar = 4)` and `Foo(int bar)` are identical signatures at the IL level. The default value is basically just a hint to compilers that support it. We want this API to take a CancellationToken, so we have it on our API. The default value is just a useful thing to have.

The issues arise when you treat optional parameters like overloads. Adding an optional parameter to an existing API is a breaking change, for example.

I'll look to see if I can find prior art, and I'm OK removing the default (which is the alternative) if people are concerned.

Reply:
It's fine and part of our guidelines